internal/infra/run.go

```diff
@@ -449,17 +452,52 @@ func pullImage(ctx context.Context, cli *client.Client, image string) error {
 	// pull image if necessary
 	if err != nil {
 		var privilegeFunc types.RequestPrivilegeFunc
-		token := os.Getenv("LOCAL_GITHUB_ACCESS_TOKEN")
-		if token != "" {
-			auth := base64.StdEncoding.EncodeToString([]byte("x:" + token))
-			privilegeFunc = func() (string, error) {
-				return "Basic " + auth, nil
+
+		if strings.HasPrefix(image, "ghcr.io/") {
+
+			token := os.Getenv("LOCAL_GITHUB_ACCESS_TOKEN")
+			if token != "" {
+				auth := base64.StdEncoding.EncodeToString([]byte("x:" + token))
+				privilegeFunc = func() (string, error) {
+					return "Basic " + auth, nil
+				}
+			} else {
+				log.Println("Failed to find credentials for GitHub container registry.")
+			}
+		} else if strings.Contains(image, ".azurecr.io/") {
+			username := os.Getenv("AZURE_REGISTRY_USERNAME")
+			password := os.Getenv("AZURE_REGISTRY_PASSWORD")
+
+			registryName := strings.Split(image, "/")[0]
+
+			if username != "" && password != "" {
+				authConfig := registry.AuthConfig{
+					Username:      username,
+					Password:      password,
+					ServerAddress: registryName,
+				}
+
+				encodedJSON, _ := json.Marshal(authConfig)
+				authStr := base64.URLEncoding.EncodeToString(encodedJSON)
+				privilegeFunc = func() (string, error) {
+					return authStr, nil
+				}
+			} else {
+				log.Println("Failed to find credentials for Azure container registry.")
 			}
+		} else {
+			log.Printf("Failed to find credentials for pulling image: %s\n.", image)
 		}
 
+		if privilegeFunc == nil {
+			return fmt.Errorf("failed to get credentials to pull image: %s", image)
+		}
```

Comment:
We want to be sure users who are using unprivileged sources still work, this looks like it will block it.